### PR TITLE
feat(agent + skill): a11y-architect agent and accessibility skill

### PR DESCRIPTION
**Pull request by gnpthbalaji**
## What Changed
- Implemented the `a11y-architect` agent and the accompanying `accessibility` skill to automate and enforce inclusive design standards.
- **Agent:** Created `inclusive-ui-agent.md` which defines the persona of a Senior Accessibility Architect responsible for auditing and planning inclusive interfaces.
- **Skill:** Created `accessibility.md` (origin: ECC) containing technical implementation logic for **WCAG 2.2 Level AA**, including specific success criteria for Web, iOS, and Android.

## Why This Change
To ensure that all new UI development follows a "Shift-Left" accessibility approach. By integrating WCAG 2.2 standards directly into the AI agent workflow:
- We automate the generation of semantic ARIA and Native accessibility traits.
- We minimize remediation debt by catching blockers (like insufficient target size or focus traps) during the design phase.
- We standardize inclusive design patterns across different platforms (Web vs. Native).

## Testing Done
- [x] Manual testing and agent validation completed
- [ ] Automated tests pass locally (`node tests/run-all.js`)
- [ ] Edge cases considered and tested

## Type of Change
- [ ] `fix:` Bug fix
- [x] `feat:` New feature
- [ ] `refactor:` Code refactoring
- [ ] `docs:` Documentation
- [ ] `test:` Tests
- [ ] `chore:` Maintenance/tooling
- [ ] `ci:` CI/CD changes

## Security & Quality Checklist
- [x] No secrets or API keys committed (ghp_, sk-, AKIA, xoxb, xoxp patterns checked)
- [ ] JSON files validate cleanly
- [ ] Shell scripts pass shellcheck (if applicable)
- [ ] Pre-commit hooks pass locally (if configured)
- [x] No sensitive data exposed in logs or output
- [x] Follows conventional commits format

## Documentation
- [x] Updated relevant documentation: Added `a11y-architect.md` and `accessibility.md`.
- [x] Added comments for complex logic: Included rationale for WCAG 2.2 specific criteria.
- [ ] README updated (if needed)

---
## Summary by cubic
Adds the `a11y-architect` agent and `accessibility` skill to enforce WCAG 2.2 Level AA across Web, iOS, and Android. Shifts accessibility left with clear workflows, cross‑platform mapping, and validation outputs.

- **New Features**
  - Added agent at `agents/a11y-architect.md` with discovery → implementation → validation and outputs: code, accessibility tree, WCAG mapping.
  - Added skill at `skills/accessibility/SKILL.md` with cross‑platform mapping (ARIA, SwiftUI traits, Compose semantics), examples, checklist, anti‑patterns, related skills, and coverage of key WCAG 2.2 items (Focus Appearance, Target Size, Redundant Entry).

- **Bug Fixes**
  - Renamed “When to activate” to “When to use” in the skill.
  - Updated iOS reference link and added the iOS HIG accessibility link.
  - Corrected terminology and examples: “Uncontained Modal Focus” (WCAG 2.1.2) replaces “Keyboard Traps”; updated Step 1 blocker example; added `[language]` placeholder to agent template code fence.
  - Removed duplicate `tools` frontmatter key in the agent spec.
  - Minor copy and formatting cleanups.

<sup>Written for commit 7f7e2c2c52bff7ac80c0691ce917c8436babb770. Summary will update on new commits.</sup>

## Summary by CodeRabbit

* **Documentation**
  * Added an Accessibility Architect specification describing a POUR-based persona, WCAG 2.2 Level AA guidance, a three-step workflow (discovery, implementation, validation), required output formats, examples, anti-patterns, and a compliance-mapping checklist.
  * Added an Accessibility skill definition detailing cross-platform implementation patterns, perceivable/operable/understandable/robust requirements, concrete Web/iOS/Android examples, validation guidance, anti-patterns, and a best-practices checklist.

### CODE REVIEW

File: agents/a11y-architect.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+Button(action: close) {
+Image(systemName: "xmark")
+.frame(width: 44, height: 44) // Standardizing hit area
+}
+.accessibilityLabel("Close modal")
+```
+
+## Reference
+
+- See skill `accessibility` to transform raw UI requirements into platform-specific accessible code (WAI-ARIA, SwiftUI, or Jetpack Compose) based on WCAG 2.2 criteria.
```
[the remark this answers is not in the document]
Comment (gnpthbalaji) [reply agents/a11y-architect.md:139]: Fixed in commit - [33673fb](https://github.com/affaan-m/everything-claude-code/commit/33673fb37a473e7f85b5f97ada85366176a8b058)

**Review by affaan-m — CHANGES_REQUESTED** [review]
Confirmed on the current PR branch.

Blocking issue:
- `agents/a11y-architect.md` declares `tools:` twice in frontmatter. YAML duplicate keys are parser-dependent; one value is effectively dead and the agent metadata is non-deterministic.

Please remove the duplicate key and keep a single `tools:` entry in the repo's standard style.

**Comment by affaan-m** [comment]
Pushed `7f7e2c2` to remove the duplicate `tools:` frontmatter key in `agents/a11y-architect.md`.

That was the blocking metadata issue called out in review. I did not bundle unrelated content changes into this push.
